semantic/tasks/config/project/install.js

```diff
@@ -0,0 +1,758 @@
+/*******************************
+            Set-up
+*******************************/
+
+var
+  fs             = require('fs'),
+  path           = require('path'),
+  defaults       = require('../defaults'),
+  release        = require('./release'),
+
+  requireDotFile = require('require-dot-file')
+;
+
+/*******************************
+          When to Ask
+*******************************/
+
+/* Preconditions for install questions */
+
+var when = {
+
+  // path
+  changeRoot: function(questions) {
+    return (questions.useRoot !== undefined && questions.useRoot !== true);
+  },
+
+  // permissions
+  changePermissions: function(questions) {
+    return (questions.changePermissions && questions.changePermissions === true);
+  },
+
+  // install
+  hasConfig: function() {
+    return requireDotFile('semantic.json', process.cwd());
+  },
+
+  allowOverwrite: function(questions) {
+    return (questions.overwrite === undefined || questions.overwrite == 'yes');
+  },
+  notAuto: function(questions) {
+    return (questions.install !== 'auto' && (questions.overwrite === undefined || questions.overwrite == 'yes'));
+  },
+  custom: function(questions) {
+    return (questions.install === 'custom' && (questions.overwrite === undefined || questions.overwrite == 'yes'));
+  },
+  express: function(questions) {
+    return (questions.install === 'express' && (questions.overwrite === undefined || questions.overwrite == 'yes'));
+  },
+
+  // customize
+  customize: function(questions) {
+    return (questions.customize === true);
+  },
+  primaryColor: function(questions) {
+    return (questions.primaryColor);
+  },
+  secondaryColor: function(questions) {
+    return (questions.secondaryColor);
+  }
+};
+
+/*******************************
+        Response Filters
+*******************************/
+
+/* Filters to user input from install questions */
+
+var filter = {
+  removeTrailingSlash: function(path) {
+    return path.replace(/(\/$|\\$)+/mg, '');
+  }
+};
+
+/*******************************
+          Configuration
+*******************************/
+
+module.exports = {
+
+  // check whether install is setup
+  isSetup: function() {
+    return when.hasConfig();
+  },
+
+  // detect whether there is a semantic.json configuration and that the auto-install option is set to true
+  shouldAutoInstall: function() {
+    var
+      config = when.hasConfig()
+    ;
+    return config['autoInstall'];
+  },
+
+  // checks if files are in a PM directory
+  getPackageManager: function(directory) {
+    var
+      // returns last matching result (avoid sub-module detection)
+      walk = function(directory) {
+        var
+          pathArray     = directory.split(path.sep),
+          folder        = pathArray[pathArray.length - 1],
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 14 spaces but found 10.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5182945)